bin/client/main.rs

```diff
@@ -100,11 +100,11 @@ async fn main() -> std::result::Result<(), ClientError> {
                 SignatureRole::Creator
             };
             // Keyfile
-            let keyfile = sign_opts.secret_file.unwrap_or_else(|| {
-                default_config_dir()
-                    .unwrap_or_else(|| PathBuf::from("."))
-                    .join("secret_keys.toml")
-            });
+            let keyfile = match sign_opts.secret_file {
+                Some(dir) => dir,
```

Comment:
Should you also ensure that this dir is created?

Reply:
I don't think so. My take is that if the user supplies a path, it is up to them to create the path. This is the way most programs work: If they use an `$XDG_*` dir, the program takes responsibility for creating (same for a hidden dir in `$HOME`). But if the user supplies a path, the tool assumes the user did the work.

In this particular case, per the Rust recommendations, we don't use `metadata()` and instead let the call to create the file be the point of failure.